firmware/quadruna/CRIT/src/app/app_mainState.c

```diff
@@ -0,0 +1,88 @@
+#include <string.h>
+#include "app_mainState.h"
+#include <stdlib.h>
+#include "app_canTx.h"
+#include "app_canRx.h"
+#include "app_canAlerts.h"
+#include "app_utils.h"
+#include "app_units.h"
+#include "app_globals.h"
+#include "io_led.h"
+#include "io_switch.h"
+
+static void mainStateRunOnTick100Hz(void)
+{
+    const bool imd_fault_latched = app_canRx_BMS_ImdLatchedFault_get();
+    io_led_enable(globals->config->imd_led, imd_fault_latched);
+
+    const bool bspd_fault_latched = app_canRx_BMS_BspdLatchedFault_get();
+    io_led_enable(globals->config->bspd_led, bspd_fault_latched);
+
+    // TODO: AMS fault
+    // const bool ams_fault_latched =
+    // io_led_enable(globals->config->ams_led, ams_fault_latched);
+
+    const bool contactors_open = app_canRx_BMS_AirNegative_get() == CONTACTOR_STATE_OPEN &&
+                                 app_canRx_BMS_AirPositive_get() == CONTACTOR_STATE_OPEN;
+    io_led_enable(globals->config->shdn_led, contactors_open);
```

Comment:
fact check with Sumrath, but I believe shdn_led should be driven by the `SDHN_SEN` pin on the MCU.